### PR TITLE
dask: `cf.read` chunks

### PR DESCRIPTION
**Pull request by davidhassell**
Add a new `chunks` parameter to `cf.read`.

### CODE REVIEW

File: cf/read_write/netcdf/netcdfread.py [hunk 1]
```diff
@@ -503,3 +510,69 @@ def _create_cfanetcdfarray(
         array = self.implementation.initialise_CFANetCDFArray(**kwargs)
 
         return array, kwargs
+
+    def _parse_chunks(self, ncvar):
+        # Parse dask chunks
```
Comment (sadielbartholomew) [inline cf/read_write/netcdf/netcdfread.py:515]: Definitely implied by the subsequent docstring!

```suggestion
```

File: cf/read_write/netcdf/netcdfread.py [hunk 2]
```diff
@@ -503,3 +510,69 @@ def _create_cfanetcdfarray(
...
+            # For ncdimensions = ('time', 'lat'):
+            #
+            # {} -> ["auto", "auto"]
+            # {'ncdim%time': 12} -> [12, "auto"]
+            # {'ncdim%time': 12, 'ncdim%lat': 10000} -> [12, 10000]
+            # {'ncdim%time': 12, 'ncdim%lat': "20MB"} -> [12, "20MB"]
+            # {'ncdim%time': 12, 'latitude': -1} -> [12, -1]
+            # {'ncdim%time': 12, 'Y': None} -> [12, None]
+            # {'ncdim%time': 12, 'ncdim%lat': (30, 90)} -> [12, (30, 90)]
+            # {'ncdim%time': 12, 'ncdim%lat': None, 'X': 5} -> [12, None]
```
Comment (sadielbartholomew) [inline cf/read_write/netcdf/netcdfread.py:553]: What is this comment block referring to, specifically? And can/should it perhaps ideally be subsumed into the docstring, as concrete examples?

Comment (sadielbartholomew) [reply cf/read_write/netcdf/netcdfread.py:553]: (For v3.14 feel free to ignore this comment, though, it's a bit fastidious.)

File: cf/read_write/read.py [hunk 3]
```diff
@@ -524,9 +523,80 @@ def read(
...
+                chunk size in bytes, preferring square-like chunk
+                shapes across all data dimensions. Any string value
+                accepted by the *chunks* parameter of the
+                `dask.array.from_array` function is permitted.
+
+                *Parameter example:*
+                  A chunksize of 2 MiB may be specified as
+                  ``'2097152'`` or ``'2 MiB'``.
+
+                If *chunks* is `-1` or `None` then each there is no
```
Comment (sadielbartholomew) [inline cf/read_write/read.py:545]:
```suggestion
                If *chunks* is `-1` or `None` then for each there is no
```

File: cf/read_write/read.py [hunk 4]
```diff
@@ -524,9 +523,80 @@ def read(
...
+                chunking, i.e. every data array has one chunk
+                regardless of its size
```
Comment (sadielbartholomew) [inline cf/read_write/read.py:547]:
```suggestion
                regardless of its size.
```

File: cf/read_write/read.py [hunk 5]
```diff
@@ -524,9 +523,80 @@ def read(
...
+                equivalent to it being defined with a value of
+                ``'auto'``.
+
+                *Parameter example:*
+                  ``{'T': '0.5 MiB', 'Y': [36, 37], 'X': None}``
+
+                *Parameter example:*
+                  If a netCDF file contains dimensions ``time``,
+                  ``z``, ``lat`` and ``lon``, then ``{'ncdim%time':
+                  12, 'ncdim%lat', None, 'ndim%lon': None}`` will
```
Comment (sadielbartholomew) [inline cf/read_write/read.py:581]:
```suggestion
                  12, 'ncdim%lat', None, 'ncdim%lon': None}`` will
```

File: cf/read_write/read.py [hunk 6]
```diff
@@ -524,9 +523,80 @@ def read(
...
+                  chunked; and all ``z`` axes are chunked to comply as
+                  closely as possible with the default chunks size.
+
+                  If the netCDF also contains a ``time`` coordinate
+                  variable with a ``standard_name`` attribute of
+                  ``'time'`` and an ``axis`` attribute of ``'T'``,
+                  then the same chunking could be specified with
+                  either ``{'time': 12, 'ncdim%lat', None, 'ndim%lon':
+                  None}`` or ``{'T': 12, 'ncdim%lat', None,
+                  'ndim%lon': None}``.
```
Comment (sadielbartholomew) [inline cf/read_write/read.py:593]:
```suggestion
                  either ``{'time': 12, 'ncdim%lat', None, 'ncdim%lon':
                  None}`` or ``{'T': 12, 'ncdim%lat', None,
                  'ncdim%lon': None}``.
```

**Review by sadielbartholomew — APPROVED** [review]
I've raised some minor comments but this is good to merge overall. (Also, we might want to add a Changelog entry to cover this new `read` keyword if we don't have one already, ready for the release.)

File: cf/read_write/netcdf/netcdfread.py (hunk 2 above)
Comment (davidhassell) [reply cf/read_write/netcdf/netcdfread.py]: Good point. It's not easily translatable to the docstring, and it is an underscore method ... but I have added a `chunks=` to each line to make it a bit clearer (https://github.com/NCAS-CMS/cf-python/pull/522/commits/22c6a2de02216cc6fae5cdf6605ae7be75c8516d). Happy to revisit after 3.14.0.

**Comment by davidhassell** [comment]
Changelog updated: https://github.com/NCAS-CMS/cf-python/pull/522/commits/2b4d93ff96beadc2564ee334cdfab1868884b0ee

**Pull request merged**